lib/auth/auth.go

```diff
@@ -738,6 +738,14 @@ var (
 		},
 	)
 
+	roleCount = prometheus.NewGauge(
```

Comment:
I wonder if a `resource_count` metric with a `kind` may be more extensible in future - but I can also see that being misleading if we're only supporting a subset of resource kinds.

Reply:
It would certainly be higher cardinality - but - I don't think to a concerning level. We probably have higher cardinality in our gRPC request/response error rate metrics etc.

Reply:
Fair enough - I think that's a reasonable enough reason to have them as distinct measures.